torchx/lib/torchx.ex

```diff
@@ -17,4 +19,69 @@ defmodule Torchx do
   """
   def device_count(:cuda), do: NIF.cuda_device_count()
   def device_count(_), do: raise("Only CUDA devices can be counted for now.")
+
+  # LibTorch API bindings
+
+  ## Creation
+
+  def arange(from, to, step \\ 1, opts \\ []) do
+    type = opts[:type] || :float
+    device = opts[:device] || :cpu
+
+    NIF.call(:arange, device, [from, to, step, type, torch_device(device)])
+    |> wrap_with_device(device)
+  end
+
+  ## Operations
+
+  def tensordot(left, right, left_axes, right_axes) do
+    {device, [left_ref, right_ref]} = to_refs([left, right])
+
+    NIF.call(:tensordot, device, [left_ref, right_ref, left_axes, right_axes])
+    |> wrap_with_device(device)
+  end
+
+  ## Utils
+
+  @doc false
+  def type_of(ref), do: NIF.scalar_type(ref) |> unwrap!()
+
+  @doc false
+  def device_of(ref),
```

Comment:
Note this should be `{device, ref}` too. But we can handle this later. :)

Reply:
Made a shortcut for now.